docs/writing_a_simple_publisher_and_subscriber.md

```diff
@@ -0,0 +1,479 @@
+# Writing a simple publisher and subscriber (Rust)
+* Goal: Create and run a publisher and subscriber node using Rust.
+* Tutorial level: Beginner
+* Time: 20 minutes
+<details><summary>Background</summary>
+
+In this tutorial you will create a pair of 
+[nodes](https://docs.ros.org/en/humble/Tutorials/Beginner-CLI-Tools/Understanding-ROS2-Nodes/Understanding-ROS2-Nodes.html) that pass information to each other via a 
+[topic](https://docs.ros.org/en/humble/Tutorials/Beginner-CLI-Tools/Understanding-ROS2-Topics/Understanding-ROS2-Topics.html) in the form of string messages. The example used here is a simple "talker" and "listener" system; one node publishes data and the other subscribes to the topic to receive that data.
+
+Since Rust doesn't have inheritance, it's not possible to inherit from `Node` as is common practice in [`rclcpp`](https://docs.ros.org/en/humble/Tutorials/Beginner-Client-Libraries/Writing-A-Simple-Cpp-Publisher-And-Subscriber.html) or [`rclpy`](https://docs.ros.org/en/humble/Tutorials/Beginner-Client-Libraries/Writing-A-Simple-Py-Publisher-And-Subscriber.html).
+
+The code used in these examples can be found [here](https://github.com/ros2-rust/ros2_rust/tree/main/examples/rust_pubsub)
+<div style="margin-left:20px;">
+<details><summary>Side-note on dependencies</summary>
+
+You may be wondering why you can't just add all your ROS 2-specific dependencies to `Cargo.toml` with `cargo add YOUR_DEPENDENCIES` and have to edit this file manually. Here is why:
```

Comment:
IMO this is also out of place this far up. This is super important information to capture, but I think this would be more appropriate when we actually talk about adding dependencies to nodes

Reply:
I think this information is important for a newcomer, as a normal Rust user who has worked with the command line tools to some extent will be confused as to why this suddenly no longer works.